File: 7-1.rb
#!/usr/bin/env ruby

crabs = File.read('7.input').strip.split(",").map(&:to_i)

leftmost = crabs.min
rightmost = crabs.max

cost = [0] * (rightmost + 1)

leftmost.upto(rightmost) do |gather|
  cost[gather] = (crabs.map { |crab| (gather - crab).abs }).sum
end

print cost.min, "\n"

File: 7-1.py
#!/usr/bin/env python

crabs = [int(crab) for crab in open('7.input').readlines()[0].split(",")]

leftmost = min(crabs)
rightmost = max(crabs)

cost = [0] * (rightmost + 1)

for gather in range(leftmost, rightmost + 1):
    cost[gather] = sum([abs(gather - crab) for crab in crabs])

print(min(cost))
